beacon-chain/db/validator.go

```diff
@@ -61,6 +61,31 @@ func (db *BeaconDB) ValidatorIndex(pubKey []byte) (uint64, error) {
 	return index, err
 }
 
+// ValidatorIndices accepts a slice of public keys and returns the corresponding map of validators indexes and public keys.
+func (db *BeaconDB) ValidatorIndices(pubKeys [][]byte) (map[uint64][]byte, error) {
+	m := make(map[uint64][]byte)
+	if !db.HasValidators(pubKeys) {
```

Comment:
Since this is "has any validator", the lookup for all validators will return an error if any of them do not exist in the database. This is a fatal condition when some validator has not been deposited yet

Reply:
changed to hasallvalidators

Reply:
it means that if not all validators have performed deposit then the validator client won't run